torch/_dynamo/eval_frame.py

```diff
@@ -364,7 +364,28 @@ def lookup_backend(compiler_fn):
     if isinstance(compiler_fn, str):
         from .optimizations import BACKENDS
 
-        compiler_fn = BACKENDS[compiler_fn]
+        if compiler_fn in BACKENDS:
+            compiler_fn = BACKENDS[compiler_fn]
+        else:
+            # See https://packaging.python.org/en/latest/guides/creating-and-discovering-plugins/#using-package-metadata
+            # for more information on using entry_points to provide plugins for python packages.
+            if sys.version_info < (3, 10):
+                try:
+                    from importlib_metadata import entry_points
+                except ImportError:
+                    raise RuntimeError(
```

Comment:
I'd suggest just silently ignoring this error rather than raising.  My thinking is: if the user doesn't have `entry_points` installed, then clearly there aren't any backends installed in it and "unknown backend" is a better error message.

IMO the "please install `entry_points`" error should come from when the backend tries to register itself (which happens first), not when we try to look it up.

Reply:
How about you just customize the error below, i.e. have something like "Cannot recognize backend: {}. Plugin mechanism for backend is disabled because importlib-metadata is not installed. Please run `pip3 install importlib-metadata` if you believe you installed external Dynamo plugin" (but only if the backend wasn't found)

Reply:
Shouldn't `pip install my-awesome-backend` depend on `entry_points` and trigger it to get installed automatically?

Reply:
@dzhulgakov Thanks for the suggestion! I agree that it is better to mention the backend name when raise the error as you said. Will update the error message.

@jansel Sorry for the confusion. The ``entry_points`` is not a package but a functionality in distributing a package. Thus, ``pip install dynamo-backend`` does not depends on ``entry_points``. Its interface gets stable in 3.10 and the querying function is implemented in a standard library ``importlib`` (see its [docs](https://docs.python.org/3/library/importlib.html)). For python < 3.10, we need package ``importlib_metadata`` to access this "future" standard library.

Reply:
I am suggesting that backend packages should contain something like this in their `setup.py`:
```py
install_requires = [...]
if sys.version_info < (3, 10):
    install_requires.append("importlib_metadata")

setup(
    ...
    install_requires=install_requires,
    ...
)
```

Reply:
> This PR will conflict with #93389 which moves/refactors this code.

Should I wait for that PR to be merged or rebase to that PR to continue?

Reply:
> I am suggesting that backend packages should contain something like this in their `setup.py`:
> 
> ```python
> install_requires = [...]
> if sys.version_info < (3, 10):
>     install_requires.append("importlib_metadata")
> 
> setup(
>     ...
>     install_requires=install_requires,
>     ...
> )
> ```
> 
> And shouldn't need to ask the user to manually install it. That way if `importlib_metadata` is missing, we know there are no backends using it installed.

I see. My concern is that the backend has no actual dependency on the ``importlib_metadata`` package as they might never use it in their codebase. 

I find that PyTorch 2.0 is going to drop the support for python 3.7. That's a good news because python introduced the ``importlib.metadata`` since python 3.8, though it is in provisional status (see the [docs](https://docs.python.org/3.8/library/importlib.metadata.html) here). Comparing with requiring the backend packages to add a new dependency, I prefer to use ``importlib.metadata`` (the usage will be slightly different for python 3.8, 3.9 vs 3.10 and onwards).

Reply:
> I see. My concern is that the backend has no actual dependency on the `importlib_metadata` package as they might never use it in their codebase.

To be useable by PyTorch+importlib_metadata, the backend packages will need to populate the necessary metadata in their setup.py.  IMO part of that metadata population should involve adding the needed dependency.  If the backends don't do specific work to define `entry_points()` it will just be empty.

Reply:
Adding the entry_points information does not require the existence of package ``importlib_metadata``. There are different ways to access the ``entry_points`` information in the current python enviroment. In very old time, people use package ``pkg_resources``. Now people use ``importlib_metadata`` (python < 3.8) and ``importlib.metadata`` (python >= 3.8). 

Since PyTorch 2.0 is going to only support python >= 3.8, I believe now there is no problem on the dependency as we can only use the standard python package ``importlib``. I have updated the PR to only use ``importlib`` and now this PR does not introduce any new dependency.

Reply:
Hi @jansel, could you have a look at this PR again? Please let me know if you have any suggestion, thank you!